### ISSUE TITLE
Connexion au LDAP

### ISSUE THREAD

**Issue by slindev** [culturecommunication/ginco#44]
Bonjour je suis bloqué à la connexion de LDAP. Avec le module d'ecplise je réussi a connecter LDAP, mais je n'arrive pas  avec la configuration du spring. 

Je modifie les 5 paramètres dans application.properties:                  ldap.base.provider.url=ldap://163.9.2.13:389/dc=irht2,dc=cnrs,dc=fr
    ldap.user.base.dn=cn=Users
    ldap.security.principal=cn=medium,cn=Users,dc=irht2,dc=cnrs,dc=fr
    ldap.security.credentials=secret
    ldap.auth.search.filter=(cn={0})

et dans applicationContext-security.xml  : 
 < ldap-server url="${ldap.base.provider.url}"
                 manager-dn="${ldap.security.principal}" manager-password="${ldap.security.credentials}" />
    < authentication-manager alias="authenticationManager" erase-credentials="false" >
        < ldap-authentication-provider
                user-search-base="${ldap.user.base.dn}"
                user-search-filter="${ldap.auth.search.filter}"
                user-context-mapper-ref="gincoUserDetailsContextMapper" />

Dans LDAP : dans dc=irht,dc=cnrs,dc=fr on a CN=Users puis CN=nom Utilisateur.
Pour chaque utilisateur, nous avons ces informations : 
objectClass, cn, objectCategory, codePage, countryCode, displayName, distinguishedName, givenName, name, objectGUID, objectSid, primaryGroupID, sAMAccountName, sAMAccountType, sn, userPrincipalName
Et j'ai cette erreur :" Erreur : Etat HTTP 500 - Unprocessed Continuation Reference(s); nested exception is javax.naming.PartialResultException: Unprocessed Continuation Reference(s); remaining name '' " , vous pourrez m'aider a débloquer ? Merci.

**Comment by fredericrolland** [comment]
bonjour,
voici notre config (je n'ai pas mis l'url exacte pour des questions de sécurité)
ldap.base.provider.url=ldap://adresse_ldap.intranet:nro_port
ldap.user.base.dn=o=gouv,c=fr
ldap.security.principal=cn=Manager,o=gouv,c=fr
ldap.security.credentials=isfet
ldap.auth.search.filter=(uid={0})

**Comment by slindev** [comment]
Oui, je vu ce que vous aviez fait, et je pense que je fait la même chose, mais j'ai " Erreur : Etat HTTP 500 - Unprocessed Continuation Reference(s); nested exception is javax.naming.PartialResultException: Unprocessed Continuation Reference(s); remaining name '' " , 

Quand je fait une recherche sur internet,Je pense que j'aurai besoin de mette le referral="follow" , mais je ne sais pas où il faut mettre. Car < ldap-server > et < ldap-authentication-provider > ne reconnaissent pas le referral.

**Comment by fredericrolland** [comment]
vous avez mis:
 ldap.base.provider.url=ldap://163.9.2.13:389/dc=irht2,dc=cnrs,dc=fr
alors que nous nous arrêtons au nro de port:
ldap.base.provider.url=ldap://adresse_ldap.intranet:nro_port

**Comment by fredericrolland** [comment]
au niveau de applicationContext-security.xml, nous avons : 
    <ldap-server url="${ldap.base.provider.url}/o=gouv,c=fr"
                 manager-dn="${ldap.security.principal}" manager-password="${ldap.security.credentials}"/>
    <authentication-manager alias="authenticationManager" erase-credentials="false">
        <ldap-authentication-provider
                user-search-base="ou=personnes" user-search-filter="uid={0}"
                user-context-mapper-ref="gincoUserDetailsContextMapper"/>
        <authentication-provider>
            <user-service>
                <user name="${default.user.login}" authorities="ROLE_ADMIN" disabled="false"
                      password="${default.user.password}"/>
                <user name="${default.expert.login}" authorities="ROLE_USER" disabled="false"
                      password="${default.expert.password}"/>
                <user name="${default.responsable.login}" authorities="ROLE_USER" disabled="false"
                      password="${default.responsable.password}"/>
            </user-service>
        </authentication-provider>
    </authentication-manager>

**Comment by slindev** [comment]
Oui, je vu ce que vous aviez fait, d'ailleurs je ne comprend pas pourquoi vous séparer le url en 2, (qu'ils sont ensembles dans application.properties ou moitier dans applicationContext-security.xml et l'autre moitier dans application.properties. A la fin, on a qu'une chaine d'url, c'est pour cela je le mis dans application.properties.) 
url="${ldap.base.provider.url}/o=gouv,c=fr"   et ldap.base.provider.url = ldap://adresse_ldap.intranet:nro_port   Donc  url="ldap://adresse_ldap.intranet:nro_port/o=gouv,c=fr " cela rassemble au mien : url="ldap://163.9.2.13:389/dc=irht2,dc=cnrs,dc=fr" 
Comme j'avais dit au début , je réussi à connecter en LDAP avec le module d'eclipse, mais avec spring j'ai cette erreur " Erreur : Etat HTTP 500 - Unprocessed Continuation Reference(s); nested exception is javax.naming.PartialResultException: Unprocessed Continuation Reference(s); remaining name '' ".

**Comment by slindev** [comment]
Je précise que nous utilisons active directory sous Windows 2012. Est- ce que vous avez déjà testé sous cette version ?

**Comment by fredericrolland** [comment]
non.
